drivers/i2c/i2c_sam0.c

```diff
@@ -493,6 +496,8 @@ static int i2c_sam0_transfer(const struct device *dev, struct i2c_msg *msgs,
 		}
 
 		if (data->msgs->flags & I2C_MSG_STOP) {
```

Comment:
In looking at this code two branches have the same exact code in them and an empty branch in the middle. This should probably be simplified to cover the case where the CMD needs to execute in a simple if() case with a comment. I understand the branches here match up with the logic but seeing the duplicate code and empty branch is so-so.

Some self documenting version of this might be...
```c
bool should_send_stop = !((data->msgs->flags & I2C_MSG_RESTART) && data->num_msgs > 1);
if (should_send_stop) {
    while (!i2c->STATUS.bit.CLKHOLD) {
    }
    i2c->CTRLB.bit.CMD = 3;
}
```

Reply:
@teburd I did make your suggested change - I just wanted to clarify that I'd be prioritising the `I2C_MSG_STOP` flag.

https://github.com/zephyrproject-rtos/zephyr/pull/41832/commits/f4e05f782b438ca8c88f79a31185d86750138949#diff-15533341d7626d04383f44f0d366a46142a634cbdd24a8ab5ef229cb3eccdbf8R503

Reply:
Ah I missed that, thanks!